internal/builder/builder.go

```diff
@@ -93,70 +92,62 @@ func constructBuilder(img imgutil.Image, newName string, metadata Metadata) (*Bu
 		return nil, err
 	}
 
-	uid, gid, err := userAndGroupIDs(img)
-	if err != nil {
-		return nil, err
+	bldr := &Builder{
+		baseImageName:       img.Name(),
+		image:               img,
+		layerWriterFactory:  layerWriterFactory,
+		metadata:            metadata,
+		lifecycleDescriptor: constructLifecycleDescriptor(metadata),
+		env:                 map[string]string{},
 	}
 
-	stackID, err := img.Label(stackLabel)
-	if err != nil {
-		return nil, errors.Wrapf(err, "get label %s from image %s", style.Symbol(stackLabel), style.Symbol(img.Name()))
-	}
-	if stackID == "" {
-		return nil, fmt.Errorf("image %s missing label %s", style.Symbol(img.Name()), style.Symbol(stackLabel))
+	if err := addImgLabelsToBuildr(bldr); err != nil {
+		return nil, errors.Wrap(err, "adding image labels to builder")
 	}
 
-	var mixins []string
-	if _, err := dist.GetLabel(img, stack.MixinsLabel, &mixins); err != nil {
-		return nil, err
+	if newName != "" && img.Name() != newName {
+		img.Rename(newName)
 	}
 
-	baseName := img.Name()
-	if newName != "" && baseName != newName {
-		img.Rename(newName)
+	return bldr, nil
+}
+
+func constructLifecycleDescriptor(metadata Metadata) LifecycleDescriptor {
```

Comment:
Similarly, to simplify the builder construction, I moved this LifecycleDescriptor creation to a separate helper func